packages/gitbook/src/app/middleware/(site)/(core)/sitemap.xml/route.ts

```diff
@@ -15,43 +16,30 @@ export const runtime = 'edge';
  */
 export async function GET(req: NextRequest) {
     const pointer = await getSiteContentPointer();
-    const { pages: rootPages } = await getSpaceContentData(pointer, pointer.siteShareKey);
 
-    const pages = flattenPages(rootPages, (page) => !page.hidden && isPageIndexable([], page));
-    const urls = await Promise.all(
-        pages.map(async ({ page, depth }) => {
-            // Decay priority with depth
-            const priority = Math.pow(2, -0.25 * depth);
-            // Normalize to keep 2 decimals
-            const normalizedPriority = Math.floor(100 * priority) / 100;
+    const { structure: siteStructure } = await getPublishedContentSite({
+        organizationId: pointer.organizationId,
+        siteId: pointer.siteId,
+        siteShareKey: pointer.siteShareKey,
+    });
 
-            const lastModified = page.updatedAt || page.createdAt;
+    // This sitemap is only available at root (/sitemap.xml).
+    if (!checkIsRootPointer(pointer, siteStructure)) {
+        return new Response('Not found', { status: 404 });
```

Comment:
Should we put a better error message:

```suggestion
        return new Response('Root sitemap is only served from the root of the site', { status: 404 });
```